src/main/java/com/github/steveice10/mc/protocol/data/MagicValues.java

```diff
@@ -268,21 +267,13 @@
         register(GameMode.CREATIVE, 1);
         register(GameMode.ADVENTURE, 2);
         register(GameMode.SPECTATOR, 3);
+        register(GameMode.UNDEFINED, 255); // https://bugs.mojang.com/browse/MC-189885
```

Comment:
I'm not aware yet exactly how this new undefined game mode works, but if it's only for the case where previous game mode is undefined in packets, would it make more sense to just make that field nullable to avoid confusion in the possible values of the current game mode?

Reply:
Vanilla behavior sends 255/-1 whenever the gamemode is undefined (usually for the new previous gamemode option). The alternative would be to do a check for gamemode 255 and then set it as null.

Reply:
I think it would be a good idea to just allow previous game mode in particular to be null to avoid adding a confusing extra value for a special case.

Reply:
Shall I add a GameMode read/write option, then, that makes it null if it detects 255?

Reply:
Yeah, it would be a good idea to just check if it's 255 and use null specifically for optional previous game mode fields.

Reply:
Fixed.